bin.src/genDefectList.py

```diff
@@ -0,0 +1,265 @@
+
+    afwDisplay.setDefaultMaskTransparency(75)
+
+# Quiet down lots of loggers, so we can see genDefectList logs better
+Log.getLogger("daf.persistence.butler").setLevel(Log.WARN)
+Log.getLogger("afw.image.Mask").setLevel(Log.WARN)
+Log.getLogger("afw.ExposureFormatter").setLevel(Log.WARN)
+Log.getLogger("daf.persistence.LogicalLocation").setLevel(Log.WARN)
+Log.getLogger("CameraMapper").setLevel(Log.WARN)
+Log.getLogger("afw.image.ExposureInfo").setLevel(Log.WARN)
+
+logger = Log.getLogger("genDefectList")
+logger.setLevel(Log.INFO if args.verbose else Log.WARN)
+if args.debug:
+    logger.setLevel(Log.DEBUG)
+
+outFileName = os.path.join(os.environ.get('OBS_PFS_DIR'),args.outFile)
+
+# Read existing defects file
+textArrRead = False
+if args.outFile != '':
+    try:
+        file = open(outFileName, 'r')
+        textArr = file.readlines()
+        textArrRead = True
+        logger.debug("Existing outFile found")
+    except IOError as e:
+        logger.debug("No existing outFile found")
+    text_file = open(outFileName, "w")
+    text_file.write("#\n")
+    text_file.write("# Defects file\n")
+    text_file.write("#\n")
+    text_file.write("# Convert to a fits table using getDefectFits.py;  this is done for you by running scons\n")
+    text_file.write("#\n")
+    text_file.write("#CCD x0    y0    width height\n")
+
+badCols = [int(item) for item in args.badCols.split(',')]
+dataDir = args.root
+if args.rerun != "":
+    dataDir = os.path.join(os.path.join(args.root, 'rerun'), args.rerun)
+butler = dafPersist.Butler( dataDir )
+
+biasesVisit = list()
+flatsVisit = list()
+flatsExpTimes = list()
+flatsLow = list()
+flatsHigh = list()
+for visit in np.arange( args.visitLow, args.visitHigh + 1 ):
+    try:
+        dataId = dict( spectrograph=args.spectrograph, arm=args.arm, visit=visit )
+        fileName = butler.get( 'raw_filename', dataId )
+        md = afwImage.readMetadata( fileName[ 0 ] )
+
+        if md.get("IMAGETYP") == "bias":
+            biasesVisit.append(visit)
+        if md.get("IMAGETYP") == "flat":
+            flatsVisit.append(visit)
+            flatsExpTimes.append(md.get('EXPTIME'))
+            if md.get('EXPTIME') == args.expTimeLow:
+                flatsLow.append(visit)
+            elif md.get('EXPTIME') == args.expTimeHigh:
+                flatsHigh.append(visit)
+    except:
+        logger.debug("visit %d not found" % visit)
+
+logger.debug("len(flatsLow) = %d" % len(flatsLow))
+logger.debug("flatsLow[0] = %d" % flatsLow[0])
+
+dataId = dict(visit=flatsLow[0], arm=args.arm, spectrograph=args.spectrograph)
+exp = butler.get('postISRCCD', dataId, immediate=True)
+image = exp.getMaskedImage().getImage().getArray()
+ccd = exp.getDetector().getId()
+logger.info("detector Id = %d" % ccd)
+
+if textArrRead:
+    firstWords = []
+    for i in range(len(textArr)):
+        firstWords.append(textArr[i].split()[0])
+    logger.debug('firstWords = %s' % firstWords)
+    for i in np.arange(len(firstWords)-1,-1,-1):
```

Comment:
Reading old defect list replaced with getting old defects with `butler.get('defects',...)` what makes this code redundant